src/transformers/models/gpt2/modeling_tf_gpt2.py

```diff
@@ -811,3 +816,164 @@ def call(
             hidden_states=transformer_outputs.hidden_states,
             attentions=transformer_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    The GPT2 Model transformer with a sequence classification head on top (linear layer).
+
+    :class:`~transformers.TFGPT2ForSequenceClassification` uses the last token in order to do the classification, as
+    other causal models (e.g. GPT-1) do.
+
+    Since it does classification on the last token, it requires to know the position of the last token. If a
+    :obj:`pad_token_id` is defined in the configuration, it finds the last token that is not a padding token in each
+    row. If no :obj:`pad_token_id` is defined, it simply takes the last value in each row of the batch. Since it cannot
+    guess the padding tokens when :obj:`inputs_embeds` are passed instead of :obj:`input_ids`, it does the same (take
+    the last value in each row of the batch).
+    """,
+    GPT2_START_DOCSTRING,
+)
+class TFGPT2ForSequenceClassification(TFGPT2PreTrainedModel, TFSequenceClassificationLoss):
```

Comment:
Be careful, `TFSequenceClassificationLoss` takes only into account `-100` as pad token id. So either you assume everywhere to have this same value, either you should redefine yourself the loss computation.

Reply:
Apparently you forgot this comment :)

Reply:
TFGPT2ForSequenceClassification uses the last token in order to do the classification. So for given task, if a pad_token_id is defined in the configuration, it finds the last token that is not a pad token in each row, If no pad_token_id is defined, it simply takes the last value in each row of the batch.

So I think TFSequenceClassificationLoss loss function should work, as it takes single last token which is not pad token. 

Let me know your views on it. 

@LysandreJik has already defined pad_token_id: 50256 for this model.
https://github.com/huggingface/transformers/issues/7493#issuecomment-706066964

Reply:
<img width="688" alt="image" src="https://user-images.githubusercontent.com/6419011/100634436-4eab0680-3355-11eb-9bef-ad657dcd0bb7.png">

This is snapshot of TFSequenceClassificationLoss, Not any mention of pad token id : -100 is aregetting ignored. 

There are other loss function like TFTokenClassificationLoss,TFNextSentencePredictionLoss where toke id : -100 is getting ignored.

Reply:
Ok, perfect for me then!